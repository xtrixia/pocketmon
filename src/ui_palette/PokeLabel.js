import { css } from "@emotion/css";
import clsx from "clsx";

import Typography from "./Typography";

import { COLORS } from "../root/colors";
import { SPACINGS } from "../root/spacings";

const labelStyle = css`
  padding: ${SPACINGS.xxs};
  border-radius: 2rem;
  text-align: center;
  min-width: 4rem;
  width: fit-content;
`;

function PokeLabel({ text, className }) {
  return (
    <Typography
      variant="subtitle"
      className={clsx(
        labelStyle,
        className,
        css`
          background: ${COLORS[text] || COLORS.normal};
          align-self: center;
        `
      )}
    >
      {text}
    </Typography>
  );
}
export default PokeLabel;
