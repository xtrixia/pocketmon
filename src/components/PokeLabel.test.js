import React from "react";
import renderer from "react-test-renderer";

import PokeLabel from "./PokeLabel";

test("PokeLabel render correctly", () => {
  const component = renderer.create(<PokeLabel text="mystic" />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toEqual("p");
  expect(tree.children[0]).toEqual("mystic");
});
