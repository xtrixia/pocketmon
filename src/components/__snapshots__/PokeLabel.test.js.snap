// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeLabel render correctly 1`] = `
<p
  className="css-1j15uqv css-fxinwe css-y31snc"
>
  mystic
</p>
`;
